File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let answer = 0;
    let queue = [0];
    
    for (let number of numbers) {
        const tempQueue = [];
        for (let sum of queue) {
            tempQueue.push(sum + number);
            tempQueue.push(sum - number);
        }
        queue = tempQueue;
    }
    for (const sum of queue) {
        if (sum === target) {
        answer++;
        }
    }

    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    n = len(numbers)
    
    def dfs(current_idx, current_number):
        # dfs의 탈출 조건 명시
        nonlocal answer
        if current_idx == n:
            if current_number == target:
                answer += 1
            return 
        dfs(current_idx + 1, current_number + numbers[current_idx])
        dfs(current_idx + 1, current_number - numbers[current_idx])
        
    dfs(0, 0)
    
    return answer